### PR TITLE
Server 4.2 docs

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
Release branch for server 4.2 docs. 

To add changes for 4.2, branch off `server-v4.2-release` and make changes to files in `_cci2/server/v4.2/`. Thank you!

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/server/v4.2/operator/manage-virtual-machines-with-vm-service.adoc [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+[#vm-provider]
+== VM provider
+The following configuration options are for the VM provider: either AWS or GCP.
+
+[#aws]
+=== AWS
+There is a section in your `values.yaml` file that configures VM Service to work with AWS EC2. During installation you will have set up a security group and authentication. See the xref:../installation/phase-3-execution-environments/#aws-vm-service[Installation Phase 3 - Execution Environments] page for more information.
+
+The informaiton in this section describes post-installation configuration options for VM service.
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/manage-virtual-machines-with-vm-service.adoc:29]:
```suggestion
The information in this section describes post-installation configuration options for VM service.
```

File: jekyll/_cci2/server/v4.2/operator/managing-orbs.adoc [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+---
+= Managing Orbs
+:page-layout: classic-docs
+:page-liquid:
+:page-description: Use this page to learn about orbs and how to manage them within CircleCI server v4.x.
+:icons: font
+:toc: macro
+:toc-title:
+
+This section describes how to manage orbs for an installation of server v4.x. Server installations include their own local orb registry. All orbs referenced in configs refer to the orbs in the server orb registry. You are responsible for maintaining orbs. This includes copying orbs from the public registry, updating orbs that may have been previously copied, and registering your company's private orbs, if they exist.
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/managing-orbs.adoc:15]: Just checking if "server v4.x" needs to be changed to the specific version or if this typically uses a variable

File: jekyll/_cci2/server/v4.2/operator/troubleshooting-and-support.adoc [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+---
+= Troubleshooting and support
+:page-layout: classic-docs
+:page-liquid:
+:page-description: Use this document to find troubleshooting steps if you are having problems with your CircleCI server v4.x installation.
+:icons: font
+:toc: macro
+:toc-title:
+
+This document describes an initial set of troubleshooting steps to take if you are experiencing problems with your CircleCI server v4.x installation. If your issue is not addressed below, you can generate a support bundle or contact your CircleCI account team.
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/troubleshooting-and-support.adoc:15]: Same comment regarding v4.x here

File: jekyll/_cci2/server/v4.2/operator/troubleshooting-and-support.adoc [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+toc::[]
+
+[#generate-support-bundle]
+== Generate support bundle
+A support bundle is used by CircleCI engineers to diagnose and fix any issues you are experiencing. They are typically requested when you open a ticket.
+
+To generate a support bundle, follow the steps below.
+
+=== Prerequisites
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/troubleshooting-and-support.adoc:25]: Missing section ID?

File: jekyll/_cci2/server/v4.2/operator/troubleshooting-and-support.adoc [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+. Install link:https://github.com/replicatedhq/troubleshoot#support-bundle[support-bundle (kubectl plugin)] to your local development machine.
+
+[source,bash]
+----
+# To install support-bundle plugin
+kubectl krew install support-bundle
+----
+
+=== Generating support bundle
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/troubleshooting-and-support.adoc:44]: Missing section ID?

File: jekyll/_cci2/server/v4.2/operator/upgrade-mongo.adoc [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+* Ensure backups have been taken. You will need a backup of MongoDB to restore to in case anything goes wrong during the upgrade progress
+* You are prepared to modify the `values.yaml`
+* `helm upgrade` will work from your system to upgrade the cluster
+* MongoDB root password is available
+
+[#upgrade]
+== Upgrade
+
+=== 1. Upgrade from MongoDB 3.6 to 4.0
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/upgrade-mongo.adoc:30]: Missing section ID?

File: jekyll/_cci2/server/v4.2/operator/upgrade-mongo.adoc [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+. Once the `helm upgrade` has completed and MongoDB has rolled, you will need to `exec` into the pod (with the root password handy) to modify the compatibility version. (Be sure to replace `<password>` with your MongoDB root password.)
++
+```bash
+kubectl exec -it mongodb-0 -- mongo -u root -p <password>
+db.adminCommand( { setFeatureCompatibilityVersion: "4.0" } )
+```
+
+. You should get a `{ "ok" : 1 }` response from Mongo. Exit out of the MongoDB shell and pod.
+
+=== 2. Upgrade from MongoDB 4.0 to 4.2
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/upgrade-mongo.adoc:59]: Missing section ID?

File: jekyll/_cci2/server/v4.2/operator/upgrade-mongo.adoc [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+. Once the `helm upgrade` has completed and MongoDB has rolled, you will need to `exec` into the pod (with the root password handy) to modify the compatibility version. (Be sure to replace `<password>` with your MongoDB root password.)
++
+```bash
+kubectl exec -it mongodb-0 -- mongo -u root -p <password>
+db.adminCommand( { setFeatureCompatibilityVersion: "4.2" } )
+```
+
+. You should get `{ "ok" : 1 }` again. Exit out of the shell and pod.
+
+=== 3. Upgrade from MongoDB 4.2 to 4.4
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/upgrade-mongo.adoc:80]: Missing section ID?

File: jekyll/_cci2/server/v4.2/operator/usage-data-collection.adoc [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+:page-liquid:
+:page-description: Learn about CircleCI usage data collection for the purpose of improving our product and services.
+:icons: font
+:toc: macro
+:toc-title:
+
+CircleCI typically collects usage data, such as logs and other aggregated data, for the purpose of improving our products and services. We never collect personally identifiable information or information that is specific to your projects or accounts.
+
+[#current-data-collected]
+## Current data collected
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/usage-data-collection.adoc:18]:
```suggestion
== Current data collected
```

File: jekyll/_cci2/server/v4.2/operator/usage-data-collection.adoc [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+Currently, server v4.x does not include the data collection service. The service will be included in a future release.
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/operator/usage-data-collection.adoc:19]: Same question re: `v4.x` (vs `v4.2`) here

File: jekyll/_cci2/server/v4.2/overview/circleci-server-overview.adoc [hunk 11]
```diff
@@ -0,0 +1,382 @@
...
+
+CircleCI server schedules CI jobs using the link:https://www.nomadproject.io/[Nomad] scheduler. The Nomad control plane runs inside of Kubernetes, while the Nomad clients, which are responsible for running scheduled CircleCI jobs, are provisioned outside the cluster. CircleCI
+server can run Docker jobs on the Nomad clients themselves or in a dedicated virtual machine (VM).
+
+Job artifacts and outputs are sent directly from jobs in Nomad to object storage (S3, Google Cloud Storage, or other supported options). Audit logs and other items from the application are also stored in object storage, so both the Kubernetes cluster and the Nomad clients need access to object storage.
+
+[#services]
+== Services
+
+CircleCI server v4.x consists of the following services. Find their descriptions and failure implications below:
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/overview/circleci-server-overview.adoc:102]: Same question re: v4.x (vs v4.2) here

File: jekyll/_cci2/server/v4.2/overview/release-notes.adoc [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+:icons: font
+:toc: macro
+:toc-title:
+
+[#overview]
+== Overview
+
+Server v4.2 introduces #TBC#
+
+[#migraiton]
```
Comment (kbatuigas) [inline jekyll/_cci2/server/v4.2/overview/release-notes.adoc:20]:
```suggestion
[#migration]
```

**Review by kbatuigas — CHANGES_REQUESTED** [review]
Mostly minor edits, a couple of (non-blocking IMO) suggestions

File: jekyll/_cci2/server/v4.2/air-gapped-installation/additional-considerations.adoc [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+----
+
+---
+
+
+
+[#service-type-load-balancers-k3s]
+== Service Type Load Balancers in K3s
+
+CircleCI server makes use of Service Type: Load Balancer kubernetes resources to listen to traffic on multiple ports. In order to function, the cluster needs to.
```
[the remark this answers is not in the document]
Comment (rosieyohannan) [reply jekyll/_cci2/server/v4.2/air-gapped-installation/additional-considerations.adoc:57]: @circlery could you take a look at this sentence?

File: jekyll/_cci2/server/v4.2/air-gapped-installation/additional-considerations.adoc [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
```
Comment (circlery) [inline jekyll/_cci2/server/v4.2/air-gapped-installation/additional-considerations.adoc:57]:
```suggestion
CircleCI server makes use of Service Type: Load Balancer kubernetes resources to listen to traffic on multiple ports.
```

Comment (circlery) [reply jekyll/_cci2/server/v4.2/air-gapped-installation/additional-considerations.adoc:57]: Let's remove this for now.

File: jekyll/_cci2/server/v4.2/air-gapped-installation/phase-1-prerequisites.adoc [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+----
+docker login cciserver.azurecr.io --username <your-circleci-provided-username> --password <your-circleci-provided-token>
+----
+
+=== b. Download all images required for this release
+Download all images required for the release of CircleCI server to your local machine. This list contains all required images for the Helm installation, the CircleCI Agent, and the Reality Check testing tool.
+
+[source, bash]
+----
+SERVER_4_1_IMAGE_LIST=`cat <<EOF
```
Comment (circlery) [inline jekyll/_cci2/server/v4.2/air-gapped-installation/phase-1-prerequisites.adoc:63]: this needs updating with the list here:
https://circleci.atlassian.net/browse/SERVER-2554

File: jekyll/_cci2/server/v4.2/air-gapped-installation/phase-1-prerequisites.adoc [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+circleci/node:latest
+circleci/python
+python:3.6.0
+EOF
+`
+----
+
+[source, bash]
+----
+echo $SERVER_4_1_IMAGE_LIST | while read -r image; do docker pull $image; done
```
Comment (circlery) [inline jekyll/_cci2/server/v4.2/air-gapped-installation/phase-1-prerequisites.adoc:136]: this also needs updating with the list in the comment above

File: jekyll/_cci2/server/v4.2/operator/circleci-server-security-features.adoc [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+* Sometimes user misuse of certain tools might accidentally print secrets to stdout which will appear in your logs. Please be aware of:
+  ** Running `env` or `printenv` which will print all your environment variables to stdout.
+  ** Literally printing secrets in your codebase or in your shell with `echo`.
+  ** Programs or debugging tools that print secrets on error.
+* Consult your VCS provider's permissions for your organization (if you are in an organization) and try to follow the link:https://en.wikipedia.org/wiki/Principle_of_least_privilege[Principle of Least Privilege].
+* Use Restricted Contexts with teams to share environment variables with a select security group. Read through the xref:../../../contexts/#restrict-a-context[contexts] document to learn more.
+* Ensure you audit who has access to SSH keys in your organization.
```
Comment (rosieyohannan) [inline jekyll/_cci2/server/v4.2/operator/circleci-server-security-features.adoc:106]:
```suggestion
* Ensure you regularly audit who has access to SSH keys in your organization.
```

**Review by kbatuigas — APPROVED** [review]

File: jekyll/_cci2/server/v4.2/overview/release-notes.adoc [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+:page-liquid:
+:page-description: Details of the new features included in each CircleCI server 4.x release.
+:icons: font
+:toc: macro
+:toc-title:
+
+[#overview]
+== Overview
+
+Server v4.2 introduces #TBC#
```
Comment (rosieyohannan) [inline jekyll/_cci2/server/v4.2/overview/release-notes.adoc:18]:
```suggestion
Server v4.2 introduces the config policies feature. Use config policies to create organization-level policies to impose rules and scopes to govern which configuration elements are required, allowed, not allowed etc.
```

**Review by circlery — APPROVED** [review]

**Pull request merged**